apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryInsertResults.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright 2020 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Apis.Bigquery.v2.Data;
+using Google.Apis.Requests;
+using System.Collections.Generic;
+using System.Linq;
+using static Google.Apis.Bigquery.v2.Data.TableDataInsertAllResponse;
+
+namespace Google.Cloud.BigQuery.V2
+{
+    /// <summary>
+    /// The results obtained after an insertion attempt.
+    /// This will contain information on invalid rows if any.
+    /// </summary>
+    public sealed class BigQueryInsertResults
+    {
+        private readonly BigQueryClient _client;
+        private readonly InsertOptions _options;
+        private readonly int _originalRowsCount;
+        private IReadOnlyList<BigQueryInsertRowErrors> _errors;
+
+        /// <summary>
+        /// Status of the insert request.
+        /// See <see cref="BigQueryInsertStatus"/> for a detailed explanation.
+        /// </summary>
+        public BigQueryInsertStatus Status
+        {
+            get
+            {
+                if (!HasErrors)
+                {
+                    return BigQueryInsertStatus.AllRowsInserted;
+                }
+                if (_errors.Count >= _originalRowsCount ||
+                   _options?.SkipInvalidRows != true)
+                {
+                    return BigQueryInsertStatus.NoRowsInserted;
+                }
+
+                return BigQueryInsertStatus.SomeRowsInserted;
+            }
+        }
+
+        /// <summary>
+        /// True if the insert request encountered errors.
+        /// Even if true, some rows may have been inserted,
+        /// see <see cref="Status"/> if you need to know the exact
+        /// outcome of the insert request.
+        /// </summary>
+        public bool HasErrors => _errors.Count > 0;
+
+        /// <summary>
+        /// Returns a sequence over rows with insert errors.
+        /// Each element represents a row with possibly multiple errors associated to it.
+        /// </summary>
+        /// <returns> A sequence over rows with insert errors.</returns>
```

Comment:
Nit: remove leading space.

Also, let's say that it will never be null. (And if that's not the case now, let's make sure it is :)